src/I18N_helper.tbasicu

```diff
@@ -0,0 +1,47 @@
+﻿#MINVERSION 1.11.1.0
+
+uses "I18N"
+DIM I18N as new cI18N
+
+'*******************************************************************
+'
+' The section below MUST be adjusted and copied to the main script
+'
+'*******************************************************************
+'#Region I18N_Tweaking
+'  Const AllLanguages_filename as String Value = "Translations" ' Translations.AllLanguages.xml
+'  Const Dev_local as String Value = "en-GB" ' In which language are written the strings in the script ?
+'  Const Additional_locals as String Value "fr-FR;it-IT" 'Which languages to prepare in the xml file ?
+'#EndRegion
+'#Include once "I18N_helper.tbasicu"
+'*******************************************************************
+'*******************************************************************
+
+
+
+I18N.Load(APP_ScriptPath, AllLanguages_filename)
+
+Function i18nh(stringValue as string, Optional stringComment as String) as String
+  Dim tempTXT as String = I18N.GetText(XMLEscape(stringValue))
+  If tempTXT = "" then
+    Return stringValue
+  else
+    Return XMLUnescapePartially(tempTXT)
+  endif
+end function
+
+Function XMLEscape(stringValue as String) as String
+  stringValue=Replace$(stringValue,Chr$(38),"&amp;") '     '&' must be first
+  stringValue=Replace$(stringValue,Chr$(34),"&quot;")
+  stringValue=Replace$(stringValue,Chr$(39),"&apos;")
+  stringValue=Replace$(stringValue,Chr$(60),"&lt;")
+  stringValue=Replace$(stringValue,Chr$(62),"&gt;")
+  Return stringValue
+end function
+
+Function XMLUnescapePartially(stringValue as String) as String
+  stringValue=Replace$(stringValue,"&lt;",Chr$(60))
```

Comment:
I would suggest to make the function symmetrical to XMLEscape, that is, to include `&apos;` for example.

Reply:
mmm, it's on purpose. In fact thoses functions do **not** act on the same part of the xml file. XMLUnescapePartially function is symmetrical to the XMLEscapePartially one that is in I18N.tbasic

Reply:
I see, thanks for the explanation 👍 I just wonder what is the purpose of partial escaping/unescaping, if we can add few more and it would be complete :)

Reply:
`<Sentence key="XMLEscape" lang="en-GB">XMLEscapePartially</Sentence>`
with XMLEscapePartially the text is more readable for translations as escaping quotes and double quotes are not required between those tags.

Reply:
I see 👍